src/pages/api/chooseValidity.ts

```diff
@@ -46,29 +73,27 @@ export default (req: NextApiRequestWithSession, res: NextApiResponse): void => {
             throw new Error('session is invalid.');
         }
 
-        if (req.body.validityInput === '0') {
-            setSession(req, res, 'The value of days your product is valid for cannot be 0.');
-            return;
-        }
+        const { validityInput, duration } = req.body;
+        const errorInfo: ErrorInfo[] = [];
 
-        if (!req.body.validityInput) {
-            setSession(req, res, 'The value of days your product is valid for cannot be empty.');
-            return;
-        }
+        errorInfo.concat(validityInputErrorCheck(errorInfo, validityInput));
 
-        if (isInvalidValidityNumber(req)) {
-            setSession(
-                req,
-                res,
-                'The value of days your product is valid for has to be a whole number between 1 and 366.',
-            );
-            return;
+        if (!duration || duration !== ('day' || 'week' || 'month' || 'year')) {
+            errorInfo.push({
+                errorMessage: 'The type of duration is needed. Choose one of the options.',
+                id: 'validity-units',
+            });
         }
+        errorArrayCheckAndRedirect(errorInfo, validityInput, duration, req, res);
```

Comment:
The way the code is already set up, could we not just call `updateSessionAttribute` the once (as all the values are already conditionally set earlier) and then conditionally redirect?

Reply:
I'm also getting an error because the code is falling into this method inside which it calls the `redirectTo` method and then doesn't call a `return`. The code is then trying to call the `redirectTo` method again on line 96.

Reply:
If you enter a return, it complains - unsure what the fix is here. Will refactor to solve.